xarray/core/missing.py

```diff
@@ -0,0 +1,258 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import Iterable
+
+import numpy as np
+import pandas as pd
+
+
+from .computation import apply_ufunc
+
+
+class BaseInterpolator(object):
+    '''gerneric interpolator class for normalizing interpolation methods'''
+    cons_kwargs = {}
+    call_kwargs = {}
+    f = None
+    kind = None
+
+    def __init__(self, xi, yi, kind=None, **kwargs):
+        self.kind = kind
+        self.call_kwargs = kwargs
+
+    def __call__(self, x):
+        return self.f(x, **self.call_kwargs)
+
+    def __repr__(self):
+        return "{type}: kind={kind}".format(type=self.__class__.__name__,
+                                            kind=self.kind)
+
+
+class NumpyInterpolator(BaseInterpolator):
+    def __init__(self, xi, yi, kind='linear', fill_value=None, **kwargs):
+        self.kind = kind
+        self.f = np.interp
+        self.cons_kwargs = kwargs
+        self.call_kwargs = {'period': self.cons_kwargs.pop('period', None)}
+
+        self._xi = xi
+        self._yi = yi
+
+        if self.cons_kwargs:
+            raise ValueError(
+                'recieved invalid kwargs: %r' % self.cons_kwargs.keys())
+
+        if fill_value is None:
+            self._left = np.nan
+            self._right = yi[-1]
```

Comment:
This is a little surprising to me -- I would expect both to default to NaN.

Reply:
Agreed. I'm not sure if this is a Pandas bug or an intention feature.

Reply:
Would it make sense to have them both default to NaN in Xarray?

Reply:
+1 let's use NaN here.